src/EventStore.Core/CertificateUtils.cs

```diff
@@ -162,12 +161,22 @@ private static bool TryReadPkcs12CertificateBundle
 					throw new Exception(
 						$"Could not find valid certificate with subject name '{certificateSubjectName}'.");
 
-				//Can this even happen?
-				if (certificates.Count > 1)
-					throw new Exception(
-						$"Could not determine a unique certificate from subject name '{certificateSubjectName}'.");
+				if (certificates.Count == 1) {
+					return certificates[0];
+				}
 
-				return certificates[0];
+				// If we get multiple of the same certificate, pick the one which expires last to allow rolling certificates
+				var certs = certificates.GetEnumerator();
+				X509Certificate2 mostRecentCert = null;
+				while (certs.MoveNext() && certs.Current is not null) {
```

Comment:
it looks like the check: `certs.Current is not null` should be moved inside the while loop - otherwise say, the enumerator returns: [null, CERT1], it'll stop after the first iteration and not consider CERT1.

Reply:
Thanks, fixed